extension-workspace-references.md

```diff
@@ -0,0 +1,115 @@
+
+### Initialization
+
+`ServerCapabilities` may contain a new field to indicate server-side support for this extension:
+
+```typescript
+interface ServerCapabilities {
+  /* ... all fields from the base ServerCapabilities ... */
+
+  /**
+   * The server provides workspace references exporting support.
+   */
+  xworkspaceReferencesProvider?: boolean;
+}
+```
+
+#### Workspace References Request
+
+The workspace references request is sent from the client to the server to export project-wide references to dependencies. That is, the response strictly returns references in the project to symbols defined in dependencies.
+
+_Request_
+* method: 'workspace/xreferences'
+* params: `WorkspaceReferencesParams` defined as follows:
+```typescript
+/**
+ * The parameters of a Workspace References Request.
+ */
+interface WorkspaceReferencesParams {
+}
+```
+
+_Response_:
+* result: `ReferenceInformation[]` defined as follows:
+```typescript
+/**
+ * Represents information about a reference to programming constructs like
+ * variables, classes, interfaces etc.
+ */
+interface ReferenceInformation {
+	/**
+	 * The location in the workspace where the `symbol` has been referenced.
+	 */
+	reference: Location;
+
+	/**
+	 * Metadata information describing the symbol being referenced.
+	 */
+	symbol: ReferenceSymbolInformation;
+}
+```
+* error: code and message set in case an exception happens during the workspace references request.
+
+Where `ReferenceSymbolInformation` is defined as follows:
+
+```typescript
+/**
+ * Represents information about a programming construct like a variable, class,
+ * interface etc that has a reference to it. Effectively, it contains data similar
+ * to SymbolInformation except all fields are optional and a metadata field is
+ * present for language-specific data.
+ *
+ * ReferenceSymbolInformation does NOT always uniquely identify a symbol. The
+ * caller should effectively consider the returned information to be
+ * information about the symbol which _generally_ (but NOT always) identifies a
+ * single symbol.
+ */
+interface ReferenceSymbolInformation {
+    /**
+     * The name of this symbol (same as `SymbolInformation.name`).
+     */
+    name?: string;
+
+    /**
+     * The kind of this symbol (same as `SymbolInformation.kind`).
+     */
+    kind?: number;
+
+    /**
+     * The URI of this symbol (same as `SymbolInformation.location.uri`).
+     */
+    uri?: string;
+
+    /**
+     * The name of the symbol containing this symbol (same as `SymbolInformation.containerName`).
+     */
+    containerName?: string;
+
+    /**
+     * The repository URI that the symbol is defined in, e.g. `github.com/golang/go`
+     * or `bitbucket.org/jespern/django-piston`.
+     */
+    repo?: string;
```

Comment:
I think we should allow this as an optional property so if the LS knows this, it can provide it - the PHP LS has info about every dependency in this form:
```json
{
            "name": "felixfbecker/advanced-json-rpc",
            "version": "v2.0.1",
            "source": {
                "type": "git",
                "url": "https://github.com/felixfbecker/php-advanced-json-rpc.git",
                "reference": "6aa733f6c0e934da848f1bd6a44b5e540de75f9d"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/felixfbecker/php-advanced-json-rpc/zipball/6aa733f6c0e934da848f1bd6a44b5e540de75f9d",
                "reference": "6aa733f6c0e934da848f1bd6a44b5e540de75f9d",
                "shasum": ""
            },
```
I think the repository URL has the same meaning as the `repository` field in package.json for example, in the most general sense it is the URL (resource locator) that allows your VCS to clone the repository.

If you have use for the commit hash or version, that would be other nice additions I could provide.